### PR TITLE
Introduce student loans eligibility model and port the QTS award year attribute

### PR DESCRIPTION
**Pull request by mec**
This is the first in as series of changes to move the eligibility of the student loans claim into it's own class. This work will invlove a lot of big changes so we want to keep the PRs as small and concise as we can to lessen the load on the reviewer.

This first PR introduces the `StudentLoans` namespace and the `Eligibility` class.

We then move on to port the `qts_award_year` from `TslrClaim` to the new eligitibility class. We will have to used some temporary hacks to keep this work green during the move, using some delegates, which will be removed once this work completes.

Along the way there have been some tightening up of tests and fixes as we encountered them.

https://trello.com/c/yt1gsap9

### CODE REVIEW

File: app/models/student_loans/eligibility.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      "2013_2014": 1,
+      "2014_2015": 2,
+      "2015_2016": 3,
+      "2016_2017": 4,
+      "2017_2018": 5,
+      "2018_2019": 6,
+      "2019_2020": 7,
+    }, _prefix: :awarded_qualified_status
+
+    validates :qts_award_year, on: [:"qts-year"], presence: {message: "Select the academic year you were awarded qualified teacher status"}
```
Comment (erbridge) [inline app/models/student_loans/eligibility.rb]: Do we no longer need to validate on `:submit`?

Comment (erbridge) [inline app/models/student_loans/eligibility.rb]: Possibly obvious Rails question, but when does this validator get called with a context?

File: app/views/claims/qts_year.html.erb [hunk 2]
```diff
@@ -1,26 +1,28 @@
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
-    <%= render("shared/error_summary", instance: current_claim, errored_field_id_overrides: {qts_award_year: "tslr_claim_qts_award_year_before_2013"}) if current_claim.errors.any? %>
+    <%= render("shared/error_summary", instance: current_claim, errored_field_id_overrides: {:"eligibility.qts_award_year" => "tslr_claim_eligibility_attributes_qts_award_year_before_2013"}) if current_claim.errors.any? %>
 
     <%= form_for current_claim, url: claim_path do |form| %>
       <%= form_group_tag current_claim do %>
+        <%= form.fields_for :eligibility, include_id: false do |fields| %>
         <h1 class="govuk-label-wrapper">
```
Comment (erbridge) [inline app/views/claims/qts_year.html.erb]: Indentation nitpick: shouldn't this all be indented another level now?

File: app/models/student_loans/eligibility.rb (hunk 1 above)
Comment (tekin) [reply app/models/student_loans/eligibility.rb]: Ah, good spot, we should definitely be validating the presence on `:submit`!

Comment (tekin) [reply app/models/student_loans/eligibility.rb]: The `ClaimUpdate` class [updates the claim in current context](https://github.com/DFE-Digital/dfe-teachers-payment-service/blob/master/app/models/claim_update.rb#L43), which gets called by the [controller](https://github.com/DFE-Digital/dfe-teachers-payment-service/blob/master/app/controllers/claims_controller.rb#L44)

File: app/views/claims/qts_year.html.erb (hunk 2 above)
Comment (tekin) [reply app/views/claims/qts_year.html.erb]: 👍🏽

Comment (mec) [reply app/views/claims/qts_year.html.erb]: 👍

File: app/models/student_loans/eligibility.rb (hunk 1 above)
Comment (erbridge) [reply app/models/student_loans/eligibility.rb]: Ah ok, and that context gets passed through to child models?

Comment (tekin) [reply app/models/student_loans/eligibility.rb]: Yep, that's right.

Comment (mec) [reply app/models/student_loans/eligibility.rb]: To the best of my knowledge custom validators have to be called explicitally with either `valid?(:context)`, 'invalid?(:context)` or `save(:context)`.

That makes our calls on line 43 in `claim_update.rb` and line 121 in `tslr_claim.rb` models.

https://guides.rubyonrails.org/active_record_validations.html#on

Looking at line 121 in the tslr claim model makes me think you are right about needing to validtate on `:submit`, feels like something we should have test coverage for but I am struggling to find any, I'll pick this up with @tekin.

**Pull request merged**